Sources/Public/DotLottie/DotLottieFile.swift

```diff
@@ -0,0 +1,152 @@
+//
+// DotLottie.swift
+// LottieFiles
+//
+// Created by Evandro Harrison Hoffmann on 27/06/2020.
+// Copyright © 2020 LottieFiles. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - DotLottieFile
+
+/// Detailed .lottie file structure
+public final class DotLottieFile {
+
+  // MARK: Lifecycle
+
+  /// Loads `DotLottie` from `URL` containing .lottie file.
+  ///
+  /// - Parameters:
+  ///  - url: URL to .lottie file
+  ///  - Returns: Deserialized `DotLottie`. Optional.
+  init(url: URL) throws {
+    fileUrl = DotLottieUtils.tempDirectoryURL.appendingPathComponent(url.deletingPathExtension().lastPathComponent)
+    guard url.isDotLottie else {
+      throw DotLottieError.invalidFileFormat
+    }
+    try decompress(from: url, to: fileUrl)
+  }
+
+  /// Loads `DotLottie` from `Data` object containing a compressed animation.
+  ///
+  /// - Parameters:
+  ///  - data: Data of .lottie file
+  ///  - filename: Name of .lottie file
+  ///  - Returns: Deserialized `DotLottie`. Optional.
+  init(data: Data, filename: String) throws {
+    fileUrl = DotLottieUtils.tempDirectoryURL.appendingPathComponent(filename.asFilename())
+    try decompress(data: data, to: fileUrl)
+  }
+
+  // MARK: Public
+
+  /// List of `LottieAnimation` in the file
+  public var animations: [LottieAnimation] = []
```

Comment:
Does this need to be public? I think it may be better to keep this internal and have consumers use the `DotLottieFile`-specific methods on `LottieAnimationView`. A `LottieAnimation` from a `DotLottieFile` isn't fully self-contained and may not play correctly without the full `DotLottieFile` context.